src/cdef.rs

```diff
@@ -114,19 +114,17 @@ fn constrain(diff: i32, threshold: i32, damping: i32) -> i32 {
   if threshold != 0 {
     let shift = cmp::max(0, damping - msb(threshold));
     let magnitude = cmp::min(diff.abs(), cmp::max(0, threshold - (diff.abs() >> shift)));
-    if diff < 0 {
-      -1 * magnitude
+
+    if diff < 0 { -magnitude } else { magnitude }
```

Comment:
I think this is fine but will mention: https://docs.rs/num/0.1.42/num/fn.signum.html